File: packages/core-php/src/RegistrationResult.php
<?php

declare(strict_types=1);

namespace OpenPasskey;

class RegistrationResult
{
    public function __construct(
        public readonly string $credentialId,
        public readonly string $publicKeyCose,
        public readonly int $signCount,
        public readonly string $rpIdHash,
        public readonly int $flags,
        public readonly bool $backupEligible,
        public readonly bool $backupState,
        public readonly string $attestationFormat,
        public readonly ?array $attestationX5C = null,
    ) {}
}

File: packages/core-ruby/lib/open_passkey/registration_result.rb
# frozen_string_literal: true

module OpenPasskey
  class RegistrationResult
    attr_reader :credential_id, :public_key_cose, :sign_count, :rp_id_hash,
                :flags, :backup_eligible, :backup_state,
                :attestation_format, :attestation_x5c

    def initialize(credential_id:, public_key_cose:, sign_count:, rp_id_hash:,
                   flags:, backup_eligible:, backup_state:,
                   attestation_format:, attestation_x5c: nil)
      @credential_id = credential_id
      @public_key_cose = public_key_cose
      @sign_count = sign_count
      @rp_id_hash = rp_id_hash
      @flags = flags
      @backup_eligible = backup_eligible
      @backup_state = backup_state
      @attestation_format = attestation_format
      @attestation_x5c = attestation_x5c
    end
  end
end
